packages/core/src/components/Form.tsx

```diff
@@ -956,10 +956,16 @@ export default class Form<
     const {
       translateString: customTranslateString,
       uiSchema = {},
-      experimental_componentUpdateStrategy = 'customDeep',
+      experimental_componentUpdateStrategy,
+      idSeparator,
+      idPrefix,
     } = this.props;
     const { schema, schemaUtils } = this.state;
     const { fields, templates, widgets, formContext, translateString } = getDefaultRegistry<T, S, F>();
+    // Use JSON.parse(), of the `JSON.stringify`'d version of the options to filter out all undefined ones
+    const globalFormOptions = JSON.parse(
+      JSON.stringify({ idPrefix, idSeparator, experimental_componentUpdateStrategy }),
+    );
```

Comment:
This will not work if #4773 adds the `nameGenerator` function prop since it couldn't be be stringified/parsed. 

Could you use lodash `omitBy`/`isUndefined`?

```suggestion
    const globalFormOptions = omitBy({ idPrefix, idSeparator, experimental_componentUpdateStrategy }, isUndefined)
```

Reply:
Let me try that!

Reply:
I went with `isNil()` rather than `isUndefined()` because we were already including it